backend/src/main/java/com/stock/admin/controller/ProductController.java

```diff
@@ -0,0 +1,109 @@
+package com.stock.admin.controller;
+
+import com.stock.admin.model.entity.Product;
+import com.stock.admin.model.request.ProductRequest;
+import com.stock.admin.model.response.ProductResponse;
+import com.stock.admin.model.response.ProductsResponse;
+import com.stock.admin.model.response.ResponseStatus;
+import com.stock.admin.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+
+/**
+ * The type Product controller.
+ */
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    @Autowired
+    private ProductService productService;
+
+    /**
+     * Gets all products.
+     *
+     * @param shopCode the shop code
+     * @return the all products
+     */
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json; charset=UTF-8")
+    @ResponseBody
+    public ProductsResponse getAllProducts(@RequestParam(required = false, name = "shopCode") String shopCode) {
+        List<Product> products;
+        if (shopCode != null) {
+            products = productService.getAllProductsByShopId(shopCode);
+        } else {
+            products = productService.getAll();
+        }
+
+        ProductsResponse productsResponse = new ProductsResponse();
```

Comment:
Rather than creating response object here, you can create and static method on Response Object to do it.

`ProductResponse` is not really required. Only `Response` is enough